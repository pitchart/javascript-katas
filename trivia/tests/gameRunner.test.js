const GameRunner = require('../src/gameRunner');

let outputs = "";
const storeLog = inputs => (outputs += inputs+"\n");

global.Math.random = () => 0.5;


describe("Game runner golden master", () => {
    beforeEach(() => {
        outputs = "";
        console["log"] = jest.fn(storeLog);
    });
    test("Game runner outputs don't break", () => {
        const gameRunner = new GameRunner();
        gameRunner.play();

        expect(outputs).toMatchSnapshot();
    });
});