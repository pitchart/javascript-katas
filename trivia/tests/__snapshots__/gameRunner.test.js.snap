// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game runner golden master Game runner outputs don't break 1`] = `
"Chet was added
They are player number 1
Pat was added
They are player number 2
Sue was added
They are player number 3
Chet is the current player
They have rolled a 4
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was correct!!!!
Chet now has 1 Gold Coins.
Pat is the current player
They have rolled a 4
Pat's new location is 4
The category is Pop
Pop Question 1
Answer was correct!!!!
Pat now has 1 Gold Coins.
Sue is the current player
They have rolled a 4
Sue's new location is 4
The category is Pop
Pop Question 2
Answer was correct!!!!
Sue now has 1 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 8
The category is Pop
Pop Question 3
Answer was correct!!!!
Chet now has 2 Gold Coins.
Pat is the current player
They have rolled a 4
Pat's new location is 8
The category is Pop
Pop Question 4
Answer was correct!!!!
Pat now has 2 Gold Coins.
Sue is the current player
They have rolled a 4
Sue's new location is 8
The category is Pop
Pop Question 5
Answer was correct!!!!
Sue now has 2 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 0
The category is Pop
Pop Question 6
Answer was correct!!!!
Chet now has 3 Gold Coins.
Pat is the current player
They have rolled a 4
Pat's new location is 0
The category is Pop
Pop Question 7
Answer was correct!!!!
Pat now has 3 Gold Coins.
Sue is the current player
They have rolled a 4
Sue's new location is 0
The category is Pop
Pop Question 8
Answer was correct!!!!
Sue now has 3 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 4
The category is Pop
Pop Question 9
Answer was correct!!!!
Chet now has 4 Gold Coins.
Pat is the current player
They have rolled a 4
Pat's new location is 4
The category is Pop
Pop Question 10
Answer was correct!!!!
Pat now has 4 Gold Coins.
Sue is the current player
They have rolled a 4
Sue's new location is 4
The category is Pop
Pop Question 11
Answer was correct!!!!
Sue now has 4 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 8
The category is Pop
Pop Question 12
Answer was correct!!!!
Chet now has 5 Gold Coins.
Pat is the current player
They have rolled a 4
Pat's new location is 8
The category is Pop
Pop Question 13
Answer was correct!!!!
Pat now has 5 Gold Coins.
Sue is the current player
They have rolled a 4
Sue's new location is 8
The category is Pop
Pop Question 14
Answer was correct!!!!
Sue now has 5 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 0
The category is Pop
Pop Question 15
Answer was correct!!!!
Chet now has 6 Gold Coins.
"
`;
